engine/table/src/main/java/io/deephaven/engine/table/impl/SelectOrUpdateListener.java

```diff
@@ -92,17 +87,8 @@ public void onUpdate(final TableUpdate upstream) {
         }
 
         analyzer.applyUpdate(acquiredUpdate, toClear, updateHelper, jobScheduler, this,
-                new SelectAndViewAnalyzer.SelectLayerCompletionHandler(allNewColumns, completedColumns) {
-                    @Override
-                    public void onAllRequiredColumnsCompleted() {
-                        completionRoutine(acquiredUpdate, jobScheduler, toClear, updateHelper);
-                    }
-
-                    @Override
-                    protected void onError(Exception error) {
-                        handleException(error);
-                    }
-                });
+                () -> completionRoutine(acquiredUpdate, jobScheduler, toClear, updateHelper),
+                this::handleException);
```

Comment:
We need protection to avoid multiple error calls. Can just add an atomic boolean and CAS. Might as well protect the completion routine, too, "just in case".